### PR TITLE
Adding logging field to virtual-node

### PR DESCRIPTION
**Pull request by kiranmeduri**
*Issue #, if available:*
#43 

*Description of changes:*
Adding logging field to virtual-node to match https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_VirtualNodeSpec.html

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/aws/appmesh.go [hunk 1]
```diff
@@ -284,6 +285,20 @@ func (c *Cloud) CreateVirtualNode(ctx context.Context, vnode *appmeshv1beta1.Vir
 		}
 	}
 
+	if vnode.Spec.Logging != nil {
+		if vnode.Spec.Logging.AccessLog != nil {
+			if vnode.Spec.Logging.AccessLog.File != nil {
```
Comment (mogren) [inline pkg/aws/appmesh.go]: Nit: I would prefer to join the if-statements, but maybe that's just me?

File: pkg/aws/appmesh.go [hunk 2]
```diff
@@ -347,6 +362,20 @@ func (c *Cloud) UpdateVirtualNode(ctx context.Context, vnode *appmeshv1beta1.Vir
...
```
Comment (mogren) [inline pkg/aws/appmesh.go]: Same as above.

File: pkg/controller/virtualnode.go [hunk 3]
```diff
@@ -211,6 +213,33 @@ func vnodeNeedsUpdate(desired *appmeshv1beta1.VirtualNode, target *aws.VirtualNo
...
+					}
+					if desired.Spec.Logging.AccessLog.File.Path != awssdk.StringValue(target.Data.Spec.Logging.AccessLog.File.Path) {
+						return true
+					}
+				}
+			}
+		}
+	} else if target.Data.Spec.Logging != nil {
+		return true
+	}
```
Comment (mogren) [inline pkg/controller/virtualnode.go]: Lots of cases here, but they all add up. Could be worth adding comments, but not a must.

Comment (kiranmeduri) [reply pkg/controller/virtualnode.go]: separated out into its own method

File: pkg/aws/appmesh.go (hunk 1 above)
Comment (kiranmeduri) [reply pkg/aws/appmesh.go]: yeah joined them, in future if there are variances then we can revisit breaking them.

**Pull request merged**